### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by BiosNod** [krakjoe/pthreads#715]
Environment
==========

  * PHP: 
PHP 7.0.5-4+donate.sury.org~xenial+1 (cli) ( ZTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
    with Zend OPcache v7.0.6-dev, Copyright (c) 1999-2016, by Zend Technologies
  * pthreads: 3.1.7dev
  * OS: 16.04.2 LTS (Xenial Xerus)

Summary
======

If there is a declaration in the foreach condition of the class inherited from Thread (foreach may even not be executed), then on all thread runs (even other classes in other places) there will be a segfault.

Reproducing Code
==============

```
<?php

# empty array
$arr = [];

echo "start foreach for empty array\n";

# foreach empty array! - never started
foreach ($arr as $el)
{
	# never show
	echo "in foreach\n";

	if ($el == 'segf')
	{
		class MyThread extends Thread
		{
			function run()
			{
				echo "ehh";
			}
		}

		$o = new MyThread;
		echo "created\n";
		$o->start();
	}
}

echo "def class AnotherThread\n";

class AnotherThread extends Thread
{
	function  run()
	{
		echo "run in AnotherThread\n";
	}
}

echo "creating object of AnotherThread\n";
$o = new AnotherThread;
echo "starting\n";
# segfault
$o->start();
# never show
echo "finished\n";
```

Expected Output
=============

start foreach for empty array
def class AnotherThread
creating object of AnotherThread
starting
finished

Actual Output
============

start foreach for empty array
def class AnotherThread
creating object of AnotherThread
starting
Segmentation fault (core dumped)

**Comment by SirSnyder** [comment]
Can not reproduce with php 7.0.12-dev on debian.

**Comment by tpunt** [comment]
I can reproduce this. PR #712 fixes this, along with potentially PR #686. I will see if I can review these tomorrow to get this fixed.

**Comment by krakjoe** [comment]
Fixed by 082f2eff01ddfbbbda3d5223a331d58c40ec6489

Thanks.
